src/ray/core_worker/task_manager.h

```diff
@@ -148,6 +153,19 @@ class TaskManager : public TaskFinisherInterface, public TaskResubmissionInterfa
                               const rpc::RayErrorInfo *ray_error_info = nullptr,
                               bool mark_task_object_failed = true) override;
 
+  /// A pending task failed. This will mark the task as failed.
```

Comment:
Mention this doesn't always mark the return object as failed depending on `mark_task_object_failed `?